internal/hcsoci/resources_lcow.go

```diff
@@ -120,6 +120,19 @@ func allocateLinuxResources(ctx context.Context, coi *createOptionsInternal, r *
 				// Mounts that map to a path in UVM are specified with 'sandbox://' prefix.
 				// example: sandbox:///a/dirInUvm destination:/b/dirInContainer
 				uvmPathForFile = mount.Source
+			} else if strings.HasPrefix(mount.Source, "hugepages://") {
+				// currently we only support 2M hugepage size
+				hugePageSubDirs := strings.Split(strings.TrimPrefix(mount.Source, "hugepages://"), "/")
+				if len(hugePageSubDirs) < 2 {
+					return errors.Errorf(`hugepages://%s' mount path is invalid, expected format: hugepages://<hugepage-size>/<hugepage-src-location>`, mount.Source)
```

Comment:
This is going to be formatted really odd. You prefix the error message with hugepages:// but if we entered this block then we already validated that `mount.Source` has this prefix so we'll get it twice. We'd get something like the below 

`hugepages://hugepages://wrongformat' mount path is invalid, expected format: hugepages://<hugepage-size>/<hugepage-src-location>`

You should be able to just print the `mount.Source` with %s and leave off the `hugepages://%s`